packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -439,42 +439,34 @@ export class ZipFS extends BasePortableFakeFS {
     if (p === null)
       throw new Error(`Unimplemented`);
 
-    let fd = this.openSync(p, `r`);
-
-    const closeStream = () => {
-      if (fd === -1)
-        return;
-      this.closeSync(fd);
-      fd = -1;
-    };
-
-    const stream = Object.assign(new PassThrough(), {
-      bytesRead: 0,
-      path: p,
-      close: () => {
-        clearImmediate(immediate);
-        closeStream();
-      },
-      _destroy: (error: Error | undefined, callback: (error?: Error) => void) => {
-        clearImmediate(immediate);
-        closeStream();
-        callback(error);
-      },
-    });
+    const fd = this.openSync(p, `r`);
+
+    const stream = Object.assign(
+      new PassThrough({
+        emitClose: true,
+        autoDestroy: true,
+        destroy: (error, callback) => {
+          clearImmediate(immediate);
+          this.closeSync(fd);
+          callback(error);
+        },
+      }),
+      {
+        close() {
+          stream.destroy();
```

Comment:
Does it support calling `destroy` multiple times? The old code seemed to tolerate it (the fd was cleared, so we weren't calling `closeSync` multiple times), the new one not.

Reply:
Yes, the stream handles that internally